sqlitecluster/SQLiteNode.cpp

```diff
@@ -1278,7 +1278,21 @@ void SQLiteNode::_onMESSAGE(SQLitePeer* peer, const SData& message) {
             peer->commandAddress = message["commandAddress"];
         }
         peer->setCommit(message.calcU64("CommitCount"), message["Hash"]);
-
+        
+        // We check the commit difference with 12,500 commits behind because that 
+        // represents ~30s of commits. If we're behind, let's close the command port
+        // so we can catch up with the cluster before processing new commands.
+        const string blockReason = "COMMITS_LAGGING_BEHIND";
+        const int64_t currentCommitDifference =  peer->commitCount - getCommitCount();
+        if (peer == _leadPeer && currentCommitDifference >= 12'500 && !_blockedCommandPort) {
+            SINFO("Node is lagging behind, closing command port so it can catch up.");
+            _server.blockCommandPort(blockReason);
+            _blockedCommandPort = true;
+        } else if (currentCommitDifference < 1'000 && _blockedCommandPort) {
+            SINFO("Node is caught up enough, unblocking command port.");
```

Comment:
Let's log the commit difference and use closing/opening instead of closing/ublocking (we also have `SuppressCommandPort`, so let's not make this more confusing)
```suggestion
            SINFO("Node is caught up enough (behind by " + SToStr(currentCommitDifference) + " commits), re-opening command port.");
```